lib/matplotlib/collections.py

```diff
@@ -2036,11 +2010,10 @@ def set_array(self, A):
                     faulty_data = True
 
             if misshapen_data:
-                _api.warn_deprecated(
-                    "3.5", message=f"For X ({width}) and Y ({height}) "
-                    f"with {self._shading} shading, the expected shape of "
-                    f"A is ({h}, {w}). Passing A ({A.shape}) is deprecated "
-                    "since %(since)s and will become an error %(removal)s.")
+                raise ValueError(
+                    f"For X ({width}) and Y ({height}) with {self._shading} "
+                    f"shading, the expected shape of A is ({h}, {w}), not "
+                    f"{A.shape}")
 
             if faulty_data:
                 raise TypeError(
```

Comment:
It seems to me that this should be a ValueError?

Reply:
Yes.